app/src/main/java/fr/free/nrw/commons/quiz/QuizChecker.java

```diff
@@ -37,7 +34,6 @@
     private boolean isUploadCountFetched;
 
     private CompositeDisposable compositeDisposable = new CompositeDisposable();
-    public Context context;
```

Comment:
Here application context is being injected by dagger. @dbrant Is having an instance of application context also considered dangerous?

In this particular class, I agree that storing the context isn't required and activity context can be used everywhere.

Reply:
> Here application context is being injected by dagger.

LeakCanary was showing a leak that pointed to the QuizChecker class, and when I looked at it, my eyes automatically jumped to the instance of `Context`.
But in fact the leak was caused by failing to dispose your CompoundDisposable, which this PR also fixes.

Technically, storing a reference to the application context isn't so bad. However, since Android (in its wisdom) doesn't differentiate between application and activity contexts, it's simply too dangerous to do this.  There are almost always ways of accomplishing what you want without keeping a reference to a Context.

Reply:
Okay, thanks for explaining. Will avoid storing a reference of the context as much as possible. :)